src/tools/ceph_dedup/ceph_dedup_daemon.cc

```diff
@@ -662,16 +668,14 @@ int make_crawling_daemon(const po::variables_map &opts)
     << ")" 
     << dendl;
 
-  std::shared_lock l(glock);
+  state = new SampleDedupWorkerThread::SampleDedupGlobal(
```

Comment:
Why is state declared outside of this function, couldn't it be a local std::unique_ptr?